src/main/java/io/confluent/connect/jdbc/JdbcSinkConnector.java

```diff
@@ -65,8 +70,38 @@ public ConfigDef config() {
 
   @Override
   public Config validate(Map<String, String> connectorConfigs) {
-    // TODO cross-fields validation here: pkFields against the pkMode
-    return super.validate(connectorConfigs);
+    /** get configuration parsed and validated individually */
+    Config config = super.validate(connectorConfigs);
+
+    configValue(config, PK_MODE)
+        .filter(cfg -> cfg.errorMessages().isEmpty()) // only if individual validation passed.
+        /**
+         * use recommended values for configuration validation
+         * {@link PrimaryKeyModeRecommender#validValues}
+         */
+        .ifPresent(pkMode -> {
+          if (!pkMode.recommendedValues().contains(pkMode.value())) {
+
+            String error = "'" + pkMode.value() + "' is invalid";
+
+            if (configValue(config, DELETE_ENABLED)
+                .map(deleteEnabled -> (Boolean) deleteEnabled.value())
+                .orElse(Boolean.FALSE)) {
+              error += ": Deletes are only supported for pk.mode 'record_key'";
+            }
+
+            pkMode.addErrorMessage(error);
+          }
+        });
+
+    return config;
```

Comment:
I'm sorry, but I'm still not convinced that we need to use the recommended values list here. It looks like we originally used that approach because it seemed reusable and could be leveraged in the future with little effort, but we've now added in special case logic just for the `pk.mode`/`delete.enabled` combination that basically renders that check redundant. Worse, it actually double-dings users by generating the same error message twice if they give a completely nonsensical primary key mode like `this_mode_does_not_exist` since they should already see a validation error from the [EnumValidator](https://github.com/confluentinc/kafka-connect-jdbc/blob/86d9377bc654603d88c1006970aa5dfa2630d434/src/main/java/io/confluent/connect/jdbc/sink/JdbcSinkConfig.java#L587-L592) that we [use for the `pk.mode` property](https://github.com/confluentinc/kafka-connect-jdbc/blob/86d9377bc654603d88c1006970aa5dfa2630d434/src/main/java/io/confluent/connect/jdbc/sink/JdbcSinkConfig.java#L397).

I don't think we should try to optimize for reusability here if we're already adding special case logic (which is, IMO, warranted in order to preserve clear error messages to users who many be just getting started with this connector).

I think we should either add a more sophisticated approach that shares the logic for valid `pk.mode`/`delete.enabled` combinations in a single place for both validation and recommended values, or acknowledge that we're only implementing preflight validation for one case right now and not try to extend the changes we make here beyond that single case.

Reply:
@C0urante thank you.

I agree. the whole thing now is very `pk.mode` focused, the half-way reusing `recommendedValues` becomes moot.